### DISCUSSION TITLE
How to collect selected row data from v-data-table in vuetify

### DISCUSSION THREAD

**Discussion by emerylin0722** [vuetifyjs/vuetify#20037] [General] [answered]
Hi 

I have a question about how to collect selected row data from v-data-table in vuetify.

My sample code is belowed. I have tried many ways but got unsuccessful result. Please assist me in solving this problem. Thanks.
![20240621_v-data-table come out undefined result](https://github.com/vuetifyjs/vuetify/assets/134043441/ce2f549e-7665-4841-8682-2843dda7c54d)
[v-data-table selected value sample code.txt](https://github.com/user-attachments/files/15921363/v-data-table.selected.value.sample.code.txt)

<template>
  <div>
    <v-data-table
      v-model="selected"
      :headers="headers"
      :items="items"
      item-value="name"
      show-select
      v-model:selected="selected"
    ></v-data-table>
    <v-btn @click="showSelected">Show Selected</v-btn>
  </div>
</template>

<script>
  import { ref } from 'vue'

  export default {
    setup() {
      const headers = [
        { text: 'Name', value: 'name' },
        { text: 'Address', value: 'address' },
        { text: 'Customer Name', value: 'cust_name' },
      ]

      const items = [
        { name: '001', address: '123 Main St', cust_name: 'John Doe' },
        { name: '002', address: '456 Elm St', cust_name: 'Jane Smith' },
      ]

      const selected = ref([])

      const showSelected = () => {
        if (selected.value.length === 0) {
          alert('No rows selected')
          return
        }

        let output = 'Selected Items:\n'
        selected.value.forEach(item => {
          output += `Key: ${item.name}, Address: ${item.address}, Customer Name: ${item.cust_name}\n`
        })
        alert(output)
      }

      return {
        headers,
        items,
        selected,
        showSelected,
      }
    },
  }
</script>

<style>
</style>

**Comment by userquin** [comment, marked as answer]
You can use v-model  with a ref array and return-object attr, check https://vuetifyjs.com/en/components/data-tables/data-and-display/#item-value search selected values

**Reply by emerylin0722** [reply]
hi userquin

Really appreciate your reply, adjust my code, it works.
